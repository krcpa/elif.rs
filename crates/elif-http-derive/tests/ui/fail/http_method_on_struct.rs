//! Test that HTTP method macros on structs fail

use elif_http_derive::get;

// This should fail - HTTP method macros should only work on functions
#[get("/test")]
pub struct BadStruct;

fn main() {}