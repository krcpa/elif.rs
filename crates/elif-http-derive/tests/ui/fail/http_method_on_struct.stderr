error: expected `fn`
 --> tests/ui/fail/http_method_on_struct.rs:7:5
  |
7 | pub struct BadStruct;
  |     ^^^^^^
